tests/test-lxd-rbac

```diff
@@ -0,0 +1,189 @@
+lxc config set rbac.api.url https://services.monstermunchkin.dev/rbac
+lxc config set core.https_address :8443
+
+# Setup remote
+(
+cat << EOF
+default-remote: local
+remotes:
+  images:
+    addr: https://images.linuxcontainers.org
+    protocol: simplestreams
+    public: true
+  local:
+    addr: unix://
+    public: false
+  rbac-user:
+    addr: https://127.0.0.1:8443
+    protocol: lxd
+    auth_type: candid
+    protocol: lxd
+    public: false
+  rbac-admin:
+    addr: https://127.0.0.1:8443
+    protocol: lxd
+    auth_type: candid
+    protocol: lxd
+    public: false
+aliases: {}
+EOF
+) > ~/snap/lxd/common/config/config.yml
+mkdir -p ~/snap/lxd/common/config/jars/
+
+mkdir -p ~/snap/lxd/common/config/servercerts/
+cp /var/snap/lxd/common/lxd/server.crt ~/snap/lxd/common/config/servercerts/rbac-user.crt
+cp /var/snap/lxd/common/lxd/server.crt ~/snap/lxd/common/config/servercerts/rbac-admin.crt
+
+# Get user credentials
+curl https://canonical-lxd.monstermunchkin.dev/config/rbac-test.user -o /var/snap/bhttp/common/rbac-test.user
+bhttp --cookiefile /var/snap/bhttp/common/rbac-user --agent /var/snap/bhttp/common/rbac-test.user https://services.monstermunchkin.dev/rbac/api/rbac/v1/service/registerable || true
+cp /var/snap/bhttp/common/rbac-user ~/snap/lxd/common/config/jars/rbac-user
+
+# Get admin credentials
+curl https://canonical-lxd.monstermunchkin.dev/config/rbac-test.admin -o /var/snap/bhttp/common/rbac-test.admin
+bhttp --cookiefile /var/snap/bhttp/common/rbac-admin --agent /var/snap/bhttp/common/rbac-test.admin https://services.monstermunchkin.dev/rbac/api/rbac/v1/service/registerable || true
+cp /var/snap/bhttp/common/rbac-admin ~/snap/lxd/common/config/jars/rbac-admin
+
+# Validate access
+echo "==> Confirming basic API access"
+lxc info rbac-user:
+lxc info rbac-admin:
+
+# Confirm initial access
+echo "==> Attempting to list instances as rbac-admin"
+lxc list rbac-admin:
+
+echo "==> Attempting to list instances as rbac-user"
+! lxc list rbac-user: || false
+
+# Create some test projects
+echo "==> Creating some test projects"
+lxc project create test-admin -c features.profiles=false
+lxc project create test-operator -c features.profiles=false
+lxc project create test-user -c features.profiles=false
+lxc project create test-auditor -c features.profiles=false
+
+# Create containers in all projects
+echo "==> Creating some instances in those projects"
+lxc launch images:alpine/edge a1 --project test-admin
+lxc launch images:alpine/edge a2 --project test-operator
+lxc launch images:alpine/edge a3 --project test-user
+lxc launch images:alpine/edge a4 --project test-auditor
+
+echo "==> Configuring rbac-user for access to projects"
+bhttp --raw --agent /var/snap/bhttp/common/rbac-test.admin https://services.monstermunchkin.dev/rbac/api/rbac/v1/resource > resources.json
+bhttp --raw --agent /var/snap/bhttp/common/rbac-test.admin https://services.monstermunchkin.dev/rbac/api/rbac/v1/user > user.json
```

Comment:
In my tests, I needed to pipe through `| cat > resources.json` to avoid `bhttp`'s snap confinement/Apparmor profile from preventing the write operation.